vsintegration/src/FSharp.Editor/Common/RoslynHelpers.fs

```diff
@@ -192,6 +192,30 @@ module internal RoslynHelpers =
             }
         Async.Start (computation, cancellationToken)
 
+    /// Execute given tasks on a background thread, running at most Environment.ProcessorCount at the same time
+    let ParallelBackgroundTasks (ct: CancellationToken) (jobs: (unit -> Task<'a>) seq) = backgroundTask {
+
+        use semaphore = new SemaphoreSlim(Environment.ProcessorCount)
```

Comment:
(really curious here) - do we have a reason to not trust the ThreadPool to size itself here?
The ThreadPool might of course let more work to be started that CPU has cores, but then the work itself might be interleaving  CPU and IO work... ?

=> I wonder if the custom SemaphoreSlim isn't defeating the purpose of a thread pool? Esp. in cases where we do not have estimates for size of "job" and the amount of CPU/IO work it might do.

Reply:
Yeah me too :) 

But subjectively the UI feels a bit more responsive this way when doing a heavy search (e.g. Range).

I think the ThreadPool can size itself, but it must spend some extra time figuring it out when we start potentially hundreds of Tasks at once. And maybe it also manages to squeeze more actual work in and less resources remain for the UI thread. And that can be a problem since there is no cancel button when you do "Find All References".